src/Lime.Transport.SignalR/ServerSignalRTransport.cs

```diff
@@ -0,0 +1,124 @@
+        private const string CLOSE_METHOD = "Close";
+        private readonly IHubContext<EnvelopeHub> _hubContext;
+        private readonly string _connectionId;
+        private readonly HubOptions _hubOptions;
+        private readonly HttpConnectionDispatcherOptions _httpConnectionDispatcherOptions;
+
+        internal ServerSignalRTransport(
+            IHubContext<EnvelopeHub> hubContext,
+            string connectionId,
+            Channel<string> envelopeChannel,
+            IEnvelopeSerializer envelopeSerializer,
+            HubOptions hubOptions,
+            HttpConnectionDispatcherOptions httpConnectionDispatcherOptions,
+
+            ITraceWriter traceWriter = null) : base(envelopeChannel,
+                                                    envelopeSerializer,
+                                                    traceWriter)
+        {
+            _hubContext = hubContext;
+            _connectionId = connectionId;
+            _hubOptions = hubOptions;
+            _httpConnectionDispatcherOptions = httpConnectionDispatcherOptions;
+        }
+
+        /// <inheritdoc />
+        public override string LocalEndPoint
+        {
+            get
+            {
+                try
+                {
+                    return _hubContext.GetLocalEndpoint(_connectionId)?.ToString();
+                }
+                catch (MissingMemberException ex)
+                {
+                    TraceWriter.TraceIfEnabledAsync(ex.ToString(), DataOperation.Error);
+                    return string.Empty;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public override string RemoteEndPoint
+        {
+            get
+            {
+                try
+                {
+                    return _hubContext.GetRemoteEndpoint(_connectionId)?.ToString();
+                }
+                catch (MissingMemberException ex)
+                {
+                    TraceWriter.TraceIfEnabledAsync(ex.ToString(), DataOperation.Error);
+                    return string.Empty;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public override IReadOnlyDictionary<string, object> Options
+        {
+            get
+            {
+                var options = new RouteValueDictionary(_httpConnectionDispatcherOptions);
+                new RouteValueDictionary(_hubOptions).ToList().ForEach(x => options[x.Key] = x.Value);
+                
+                return options;
+            }
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="InvalidOperationException">Thrown when the transport is not connected.</exception>
+        public override async Task SendAsync(Envelope envelope, CancellationToken cancellationToken)
+        {
+            ThrowIfNotConnected();
+
+            await base.SendAsync(envelope, cancellationToken).ConfigureAwait(false);
+
+            string envelopeSerialized = EnvelopeSerializer.Serialize(envelope);
```

Comment:
Naming, always put the source class name in the end. Example: serialized**Envelope**, request**Command**, etc.

```suggestion
            string serializedEnvelope = EnvelopeSerializer.Serialize(envelope);
```